odl/discr/grid.py

```diff
@@ -681,9 +684,46 @@ def append(self, other):
         See Also
         --------
         insert
+        extend
         """
         return self.insert(self.ndim, other)
 
+    def extend(self, seq):
+        """Return a copy of ``self``, extended by all members of ``seq``.
+
+        Parameters
+        ----------
+        seq : sequence of `RectGrid`
```

Comment:
Decent suggestion, also because it makes one out of two methods look less similar to the corresponding list method. It would then also make sense to do the same for `insert`, still with a single index, so one can add a bunch of stuff at a given position.